test/controllers/CorrespondenceAddressControllerSpec.scala

```diff
@@ -78,6 +78,8 @@ class CorrespondenceAddressControllerSpec extends PlaySpec with GuiceOneServerPe
             document.getElementById("postcode").text() must be("")
             document.getElementById("country_field").text() must include("Country")
             document.getElementById("submit").text() must be("Continue")
+            document.getElementById("backLinkHref").text() must be("Back")
+            document.getElementById("backLinkHref").attr("href") must be("/ated-subscription/registered-business-address")
```

Comment:
Also need to add test for other scenario (when editing existing data) e.g. by adding 
```
            document.getElementById("backLinkHref").text() must be("Back")
            document.getElementById("backLinkHref").attr("href") must be("/ated-subscription/review-business-details")
```
to the "if data exists in keystore, fill the form with that data in the page" test

Reply:
I've added this